cdap-api-common/src/main/java/co/cask/cdap/internal/io/SchemaTypeAdapter.java

```diff
@@ -261,6 +305,36 @@ private Schema readRecord(JsonReader reader, Map<String, Schema> knownRecords) t
     return schema;
   }
 
+  private List<Schema.Field> getFields(String recordName, JsonReader reader, Map<String, Schema> knownRecords)
+    throws IOException {
+    knownRecords.put(recordName, null);
+    List<Schema.Field> fieldBuilder = new ArrayList<>();
+    reader.beginArray();
+    while (reader.peek() != JsonToken.END_ARRAY) {
+      reader.beginObject();
+      String fieldName = null;
+      Schema innerSchema = null;
+
+      while (reader.hasNext()) {
+        String name = reader.nextName();
+        switch(name) {
+          case "name":
+            fieldName = reader.nextString();
+            break;
+          case "type":
+            innerSchema = read(reader, knownRecords);
+            break;
+          default:
+            reader.skipValue();
```

Comment:
But was that the intention? I thought the jira specifically mentions only handling avro schema, and not ignoring unknown fields? See the first comment in https://issues.cask.co/browse/CDAP-5907